File: examples/go/string_operations.go
package main

import "fmt"

func reverse(s string) string {
    runes := []rune(s)
    for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
        runes[i], runes[j] = runes[j], runes[i]
    }
    return string(runes)
}

func countVowels(s string) int {
    count := 0
    vowels := "aeiouAEIOU"
    for _, char := range s {
        for _, vowel := range vowels {
            if char == vowel {
                count = count + 1
                break
            }
        }
    }
    return count
}

func isPalindrome(s string) bool {
    reversed := reverse(s)
    return s == reversed
}

func main() {
    fmt.Println("String Processing Demo")
    
    text := "hello"
    fmt.Printf("Original: %s\n", text)
    fmt.Printf("Reversed: %s\n", reverse(text))
    fmt.Printf("Vowel count: %d\n", countVowels(text))
    fmt.Printf("Is palindrome: %t\n", isPalindrome(text))
    
    // Test with palindrome
    palindrome := "racecar"
    fmt.Printf("\nPalindrome test: %s\n", palindrome)
    fmt.Printf("Is palindrome: %t\n", isPalindrome(palindrome))
    
    // String array
    words := [3]string{"go", "programming", "language"}
    fmt.Println("\nWords and their vowel counts:")
    for i := 0; i < 3; i++ {
        word := words[i]
        vowels := countVowels(word)
        fmt.Printf("%s: %d vowels\n", word, vowels)
    }
}

File: examples/python/string_operations.py
def reverse_string(s):
    return s[::-1]

def count_vowels(s):
    count = 0
    vowels = "aeiouAEIOU"
    for char in s:
        if char in vowels:
            count = count + 1
    return count

def is_palindrome(s):
    reversed_s = reverse_string(s)
    return s == reversed_s

def main():
    print("String Processing Demo")
    
    text = "hello"
    print("Original:", text)
    print("Reversed:", reverse_string(text))
    print("Vowel count:", count_vowels(text))
    print("Is palindrome:", is_palindrome(text))
    
    palindrome = "racecar"
    print()
    print("Palindrome test:", palindrome)
    print("Is palindrome:", is_palindrome(palindrome))
    
    words = ["go", "programming", "language"]
    print()
    print("Words and their vowel counts:")
    for word in words:
        vowels = count_vowels(word)
        print(word + ":", vowels, "vowels")

if __name__ == "__main__":
    main()